File: FSE_ProjectManager/PM.Utilities/Filter/SortDescriptor.cs
namespace PM.Utilities.Filter
{
    public enum SortDirection
    {
        Undefined = 0,
        ASC = 1,
        DSC = 2
    }

    public class SortDescriptor
    {
        /// <summary>
        /// The field that is sorted.
        /// </summary>
        public string Field { get; set; }

        public string Dir { get; set; }

        /// <summary>
        /// The sort direction.
        ///
        /// If no direction is set, the descriptor will be skipped during processing.
        /// </summary>
        public SortDirection Direction
        {
            get
            {
                if (!string.IsNullOrEmpty(Dir))
                {
                    switch (Dir.Trim().ToLower())
                    {
                        case "asc":
                            return SortDirection.ASC;

                        case "desc":
                            return SortDirection.DSC;

                        default:
                            throw new System.InvalidOperationException();
                    }
                }

                return SortDirection.Undefined;
            }
        }
    }
}

File: FSE_ProjectManager_UI/src/app/shared/filter/models/sort-descriptor.ts
/**
 * The sort descriptor used by the `orderBy` method.
 *
 * It has the following properties:
 */
export interface SortDescriptor {
  /**
   * The field that is sorted.
   */
  field: string;
  /**
   * The sort direction.
   * If no direction is set, the descriptor will be skipped during processing.
   *
   * The available values are:
   * - `asc`
   * - `desc`
   */
  dir?: 'asc' | 'desc';
}
